### PR TITLE
refactor: Refactor and rework some of the common logic from HIP-904 system contracts

### LINKED ISSUE #16214: Refactor HIP-904 technical debt
Refactor the copy code and extract to helper classes.
This applies to SC logic and e2e tests.
Apply the refactored `isSelectorIfConfigEnabled` method and remove the Deprecated one.

### PR DESCRIPTION
**Pull request by stoyanov-st**
**Description**:
This PR aims to clear the technical debt from the newly implemented HIP-904 system contract functions.
This includes:
- Extracting common logic from the Decoder classes in helper class
- Reusing the refactored `isSelectorIfConfigEnabled` method (it can now take varargs for the function selectors)
- Reworking some unit tests to increase the coverage
- Reworking e2e tests and extracting the common logic in helper class
- Combine the HRC HIP-904 contracts into a single HRC904.sol.

**Related issue(s)**:

Fixes #16214 

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by lukelee-sl — DISMISSED** [review]
LGTM.  Seems like quite a sensible refactor to me.  Thanks for doing this to add clarity.  I had one small question that is more of a stylistic matter.  Address it only if you think it is important.

File: hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/utils/PendingAirdropsHelper.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+import com.hedera.hapi.node.base.NftID;
+import com.hedera.hapi.node.base.PendingAirdropId;
+import com.hedera.hapi.node.base.TokenID;
+import com.hedera.hapi.node.base.TokenType;
+import com.hedera.node.app.service.contract.impl.exec.systemcontracts.hts.HtsCallAttempt;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.Arrays;
+import java.util.List;
+
+public final class PendingAirdropsHelper {
```
[the remark this answers is not in the document]
Comment (stoyanov-st) [reply hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/utils/PendingAirdropsHelper.java:35]: The cancel/claim decoder tests are verifying for it. This is just their common logic extracted and reused.

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/precompile/airdrops/SystemContractAirdropHelper.java [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+    public static GetBalanceOperation checkForEmptyBalance(
+            final SpecAccount receiver, final List<SpecFungibleToken> tokens, final List<SpecNonFungibleToken> nfts) {
+        return receiver.getBalance().andAssert(balance -> {
+            tokens.forEach(token -> balance.hasTokenBalance(token.name(), 0L));
+            nfts.forEach(nft -> balance.hasTokenBalance(nft.name(), 0L));
+        });
+    }
+
+    public static Address[] prepareSenderAddresses(@NonNull HapiSpec spec, @NonNull SpecAccount... senders) {
```
[the remark this answers is not in the document]
Comment (stoyanov-st) [reply hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/precompile/airdrops/SystemContractAirdropHelper.java]: Good point, thanks. Change added.

**Review by lukelee-sl — APPROVED** [review]
LGTM

**Pull request merged**